core/src/eval/operation.rs

```diff
@@ -707,6 +707,10 @@ impl<R: ImportResolver, C: Cache> VirtualMachine<R, C> {
                             env: Environment::new(),
                         })
                     }
+                    Term::EnumVariant { arg, .. } => Ok(Closure {
+                        body: seq_terms(std::iter::once(arg), pos_op),
+                        env,
+                    }),
```

Comment:
Just to make sure: enum tags are already in normal form, so we didn't have to worry about them before, right?

Reply:
Right, enum tags are considered weak head normal forms, the same as any compound datastructure.  They are the structural version of a type constructor, and in Haskell for example, type constructors are indeed WHNF.